lib/view_component/compiler.rb

```diff
@@ -85,7 +104,21 @@ def define_render_template_for
         "elsif variant.to_sym == :#{variant}\n    #{call_method_name(variant)}"
       end.join("\n")
 
-      component_class.class_eval <<-RUBY, __FILE__, __LINE__ + 1
+      if development?
+        component_class.class_eval <<-RUBY, __FILE__, __LINE__ + 1
+        def render_template_for(variant = nil)
```

Comment:
Might we be able to extract the shared parts of this heredoc? I could see these two branches falling out of sync.

Reply:
Good idea. Done!